### ISSUE TITLE
Segmentation fault using ByteStore operations

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Bytes#186]
The following test produces a segmentation fault:

```
    @Test
    public void segFault() {
        {
            byte[] arr = new byte[4];
            BytesStore bs = BytesStore.wrap(arr);
            bs.writeInt(0, 0xaaaaaaaa);
            //System.exit(0);
        }

        BytesStore bs = BytesStore.from("this is a test          ");
        bs.append(14, 16, 12);
        bs.isClear();
        bs.writeUtf8(14, "this is a another text it should over write the other");
    }


#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000011a7a71b5, pid=89308, tid=0x0000000000001103
#
# JRE version: OpenJDK Runtime Environment (8.0_252-b09) (build 1.8.0_252-b09)
# Java VM: OpenJDK 64-Bit Server VM (25.252-b09 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# j  net.openhft.chronicle.bytes.BytesInternal.append8bit(JLnet/openhft/chronicle/bytes/RandomDataOutput;Ljava/lang/CharSequence;II)V+100
#

```

[hs_err_pid89308.log](https://github.com/OpenHFT/Chronicle-Bytes/files/6316541/hs_err_pid89308.log)

**Comment by minborg** [comment]
Related to #187

**Comment by peter-lawrey** [comment]
These classes should really only be used if you know what you are doing. i.e. probably should have been internal.

**Comment by peter-lawrey** [comment]
Can we mark all BytesStore to move to internal in the next version x.22

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.21ea45](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.21ea45), [BOM-2.21ea113](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.21ea113)
